src/logic/BrazeMessages.tsx

```diff
@@ -123,13 +159,17 @@ class BrazeMessages implements BrazeMessagesInterface {
 
     getMessageForBanner(): Promise<BrazeMessage> {
         // If there's already a message in the cache, return it
-        const messageFromCache = this.cache.peek('Banner', this.appboy, this.errorHandler);
+        const messagesFromCache = this.cache.all('Banner', this.appboy, this.errorHandler);
 
-        if (messageFromCache) {
+        const [firstValidMessage] = messagesFromCache.filter((msg) =>
```

Comment:
What do you think of the name `firstRenderableMessage`? Just to be really clear what we mean here.

Reply:
Done!